src/backend/marsha/core/api/video.py

```diff
@@ -465,6 +477,102 @@ def update_live_state(
 
         return Response({"success": True})
 
+    @action(
+        methods=["get"],
+        detail=True,
+        url_path="pairing-secret",
+        permission_classes=[
+            permissions.IsTokenResourceRouteObject
+            & (permissions.IsTokenInstructor | permissions.IsTokenAdmin)
+        ],
+    )
+    # pylint: disable=unused-argument
+    def pairing_secret(
+        self,
+        request,
+        pk=None,
+    ):
+        """Generate a secret for pairing an external device to a live stream.
+
+        Parameters
+        ----------
+        request : Type[django.http.request.HttpRequest]
+            The request on the API endpoint
+        pk: string
+            The primary key of the video
+
+        Returns
+        -------
+        Type[rest_framework.response.Response]
+            HttpResponse with the generated secret.
+        """
+        video = self.get_object()
+        if video.live_type != JITSI:
+            return Response(
+                {"detail": "Matching video is not a Jitsi Live."}, status=400
+            )
+
+        try:
+            live_pairing = LivePairing.objects.get(video=video)
+        except LivePairing.DoesNotExist:
+            live_pairing = LivePairing(video=video)
+        live_pairing.generate_secret()
+        live_pairing.save()
+
+        serializer = serializers.LivePairingSerializer(instance=live_pairing)
+        return Response(serializer.data)
+
+
+class PairingChallengeThrottle(AnonRateThrottle):
+    """Throttling for pairing challenge requests."""
+
+    def get_ident(self, request):
+        """Identify requests by box_id instead of IP"""
+        return request.data.get("box_id")
+
+
+@api_view(["POST"])
+@throttle_classes([PairingChallengeThrottle])
+def pairing_challenge(request):
+    """View handling pairing challenge request to stream from an external device.
+
+    Parameters
+    ----------
+    request : Type[django.http.request.HttpRequest]
+        The request on the API endpoint, it must contains a payload with the following fields:
+            - box_id: uuid from external device.
+            - secret: generated pairing secret.
+
+    Returns
+    -------
+    Type[rest_framework.response.Response]
+        HttpResponse containing live video credentials.
+
+    """
+
+    serializer = serializers.PairingChallengeSerializer(data=request.data)
+    if not serializer.is_valid():
+        return Response(serializer.errors, status=400)
+
+    try:
+        live_pairing = LivePairing.objects.get(secret=request.data.get("secret"))
```

Comment:
If I'm not mistaken you forgot to use the box I'd so you are not blocking unknown boxes from making many requests...

Reply:
Thanks to @lunika , I added a test which should cover this case : `test_api_video_pairing_challenge_unknown_device_throttling` WDYT ?

Reply:
Looking at tests, are we ok with this ?